listings/applications/campaign_factory/src/contract.cairo

```diff
@@ -0,0 +1,153 @@
+    use starknet::{
+        ContractAddress, ClassHash, SyscallResultTrait, syscalls::deploy_syscall,
+        get_caller_address, get_contract_address
+    };
+    use alexandria_storage::list::{List, ListTrait};
+    use campaign::campaign::{ICampaignDispatcher, ICampaignDispatcherTrait};
+    use components::ownable::ownable_component;
+
+    component!(path: ownable_component, storage: ownable, event: OwnableEvent);
+
+    #[abi(embed_v0)]
+    impl OwnableImpl = ownable_component::Ownable<ContractState>;
+    impl OwnableInternalImpl = ownable_component::OwnableInternalImpl<ContractState>;
+
+    #[storage]
+    struct Storage {
+        #[substorage(v0)]
+        ownable: ownable_component::Storage,
+        /// Store all of the created campaign instances' addresses
+        campaigns: List<ICampaignDispatcher>,
+        /// Store the class hash of the contract to deploy
+        campaign_class_hash: ClassHash,
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    pub enum Event {
+        #[flat]
+        OwnableEvent: ownable_component::Event,
+        ClassHashUpdated: ClassHashUpdated,
+        ClassHashUpdateFailed: ClassHashUpdateFailed,
+        CampaignCreated: CampaignCreated,
+    }
+
+    #[derive(Drop, starknet::Event)]
+    pub struct ClassHashUpdated {
+        pub new_class_hash: ClassHash,
+    }
+
+    #[derive(Drop, starknet::Event)]
+    pub struct ClassHashUpdateFailed {
+        pub campaign: ContractAddress,
+        pub errors: Array<felt252>
+    }
+
+    #[derive(Drop, starknet::Event)]
+    pub struct CampaignCreated {
+        pub caller: ContractAddress,
+        pub contract_address: ContractAddress
+    }
+
+    pub mod Errors {
+        pub const CLASS_HASH_ZERO: felt252 = 'Class hash cannot be zero';
+    }
+
+    #[constructor]
+    fn constructor(ref self: ContractState, class_hash: ClassHash) {
+        assert(class_hash.is_non_zero(), Errors::CLASS_HASH_ZERO);
+        self.campaign_class_hash.write(class_hash);
+        self.ownable._init(get_caller_address());
+    }
+
+
+    #[abi(embed_v0)]
+    impl CampaignFactory of super::ICampaignFactory<ContractState> {
+        // ANCHOR: deploy
+        fn create_campaign(
+            ref self: ContractState,
+            title: ByteArray,
+            description: ByteArray,
+            target: u256,
+            duration: u64,
+            token_address: ContractAddress,
+        ) -> ContractAddress {
+            let caller = get_caller_address();
+
+            // Create contructor arguments
+            let mut constructor_calldata: Array::<felt252> = array![];
+            ((caller, title, description, target), duration, token_address)
+                .serialize(ref constructor_calldata);
+
+            // Contract deployment
+            let (contract_address, _) = deploy_syscall(
+                self.campaign_class_hash.read(), 0, constructor_calldata.span(), false
+            )
+                .unwrap_syscall();
```

Comment:
Don't forgot that you are deploying as the factory so the owner of the campaigns will be the factory itself.
Be sure to either delegating the deploy so that the owner of the campaigns contract is not the factory but the caller, or add each `onlyOwner` entrypoints of the campaign contract inside the factory contract.

Reply:
I updated the code to make the roles clearer:
- `owner` is now the factory
- `creator` is the caller who initiated the creation of the contract through the factory